tests/test_nemotron_prompt_modes.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Tests for Nemotron prompt format and structure.
+
+This module verifies that:
+1. Nemotron models use message-based prompts from nemotron.yml
+2. DeepSeek models use content-based prompts from deepseek.yml
+3. Some tasks in nemotron.yml have "detailed thinking" for internal steps
+   (generate_bot_message and generate_value)
+4. Other tasks (generate_user_intent and generate_next_steps) in nemotron.yml don't have "detailed thinking"
+"""
+
+import yaml
+
+from nemoguardrails import RailsConfig
+from nemoguardrails.llm.prompts import _get_prompt, _prompts, get_prompt
+from nemoguardrails.llm.types import Task
+
+NEMOTRON_MODEL = "nvidia/llama-3.1-nemotron-ultra-253b-v1"
```

Comment:
let's check both a reasoning nemotron, like you have here, and that a strictly non-reasoning one like an older nemotron uses what we expect.